sdk/spring/azure-spring-boot-starter-keyvault-certificates/src/main/java/com/azure/spring/security/keyvault/certificates/starter/KeyVaultCertificatesEnvironmentPostProcessor.java

```diff
@@ -3,112 +3,89 @@
 
 import com.azure.security.keyvault.jca.KeyVaultJcaProvider;
+import com.azure.security.keyvault.jca.KeyVaultKeyStore;
 import com.azure.security.keyvault.jca.KeyVaultTrustManagerFactoryProvider;
-
-import java.security.Security;
-import java.util.Properties;
-import javax.net.ssl.HttpsURLConnection;
-
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.env.EnvironmentPostProcessor;
 import org.springframework.core.annotation.Order;
 import org.springframework.core.env.ConfigurableEnvironment;
 import org.springframework.core.env.MutablePropertySources;
 import org.springframework.core.env.PropertiesPropertySource;
+import org.springframework.util.StringUtils;
 
-import static org.springframework.core.Ordered.LOWEST_PRECEDENCE;
+import javax.net.ssl.HttpsURLConnection;
+import java.security.Security;
+import java.util.Optional;
+import java.util.Properties;
 
 /**
  * Leverage {@link EnvironmentPostProcessor} to add Key Store property source.
  */
-@Order(LOWEST_PRECEDENCE)
+@Order
 public class KeyVaultCertificatesEnvironmentPostProcessor implements EnvironmentPostProcessor {
 
     @Override
-    public void postProcessEnvironment(ConfigurableEnvironment environment,
-            SpringApplication application) {
-
-        Properties systemProperties = System.getProperties();
+    public void postProcessEnvironment(ConfigurableEnvironment environment, SpringApplication application) {
 
-        String uri = environment.getProperty("azure.keyvault.uri");
-        if (uri != null) {
-            systemProperties.put("azure.keyvault.uri", uri);
-
-            String tenantId = environment.getProperty("azure.keyvault.tenant-id");
-            if (tenantId != null) {
-                systemProperties.put("azure.keyvault.tenant-id", tenantId);
-            }
-            
-            String aadAuthenticationUrl = environment.getProperty("azure.keyvault.aad-authentication-url");
-            if (aadAuthenticationUrl != null) {
-                systemProperties.put("azure.keyvault.aad-authentication-url", aadAuthenticationUrl);
-            }
-
-            String clientId = environment.getProperty("azure.keyvault.client-id");
-            if (clientId != null) {
-                systemProperties.put("azure.keyvault.client-id", clientId);
-            }
-
-            String clientSecret = environment.getProperty("azure.keyvault.client-secret");
-            if (clientSecret != null) {
-                systemProperties.put("azure.keyvault.client-secret", clientSecret);
-            }
+        if (environment.getProperty("azure.keyvault.uri") == null) {
+            return;
+        }
 
-            String managedIdentity = environment.getProperty("azure.keyvault.managed-identity");
-            if (managedIdentity != null) {
-                systemProperties.put("azure.keyvault.managed-identity", managedIdentity);
+        putEnvironmentPropertyToSystemProperty(environment, "azure.keyvault.aad-authentication-url");
```

Comment:
better add a comment here to explain why copy the env.prop to system properties.

Reply:
Added in the method.